code/modules/projectiles/guns/projectile.dm

```diff
@@ -51,21 +52,34 @@
 		return 0
 	return 1
 
+/obj/item/gun/projectile/proc/can_reload()
+	return !magazine
+
+/obj/item/gun/projectile/proc/reload(obj/item/ammo_box/magazine/AM, mob/user as mob)	
+		user.remove_from_mob(AM)
+		magazine = AM
+		magazine.loc = src
+		playsound(src, magin_sound, 50, 1)
+		chamber_round()
+		AM.update_icon()
+		update_icon()
+		return
+
 /obj/item/gun/projectile/attackby(var/obj/item/A as obj, mob/user as mob, params)
 	..()
 	if(istype(A, /obj/item/ammo_box/magazine))
 		var/obj/item/ammo_box/magazine/AM = A
-		if(!magazine && istype(AM, mag_type))
-			user.remove_from_mob(AM)
-			magazine = AM
-			magazine.loc = src
-			to_chat(user, "<span class='notice'>You load a new magazine into \the [src].</span>")
-			chamber_round()
-			A.update_icon()
-			update_icon()
-			return 1
-		else if(magazine)
-			to_chat(user, "<span class='notice'>There's already a magazine in \the [src].</span>")
+		if(istype(AM, mag_type))
+			if(can_reload())
+				reload(AM, user)
+				to_chat(user, "<span class='notice'>You load a new magazine into \the [src].</span>")
+				return TRUE
+			else if(!can_reload() && !can_tactical)
+				to_chat(user, "<span class='notice'>There's already a magazine in \the [src].</span>")
+				return TRUE

```

Comment:
Making an else here to tactically reload might be a good idea. Due to you including can_tactical in the projectile class